src/db.ts

```diff
@@ -0,0 +1,148 @@
+        this.postgres = postgres
+        this.kafkaProducer = kafkaProducer
+        this.clickhouse = clickhouse
+    }
+
+    public async postgresQuery<R extends QueryResultRow = any, I extends any[] = any[]>(
+        queryTextOrConfig: string | QueryConfig<I>,
+        values?: I
+    ): Promise<QueryResult<R>> {
+        return this.postgres.query(queryTextOrConfig, values)
+    }
+
+    public async fetchPerson(teamId: number, distinctId: string): Promise<Person | undefined> {
+        const selectResult = await this.postgresQuery(
+            `SELECT
+                posthog_person.id, posthog_person.created_at, posthog_person.team_id, posthog_person.properties,
+                posthog_person.is_user_id, posthog_person.is_identified, posthog_person.uuid,
+                posthog_persondistinctid.team_id AS persondistinctid__team_id,
+                posthog_persondistinctid.distinct_id AS persondistinctid__distinct_id
+            FROM posthog_person
+            JOIN posthog_persondistinctid ON (posthog_persondistinctid.person_id = posthog_person.id)
+            WHERE
+                posthog_person.team_id = $1
+                AND posthog_persondistinctid.team_id = $1
+                AND posthog_persondistinctid.distinct_id = $2`,
+            [teamId, distinctId]
+        )
+        const rawPerson: RawPerson = selectResult.rows[0]
+        return { ...rawPerson, created_at: DateTime.fromISO(rawPerson.created_at) }
+    }
+
+    public async createPerson(
+        createdAt: DateTime,
+        properties: Properties,
+        teamId: number,
+        isUserId: number | null,
+        isIdentified: boolean,
+        uuid: string
+    ): Promise<Person> {
+        const insertResult = await this.postgresQuery(
+            'INSERT INTO posthog_person (created_at, properties, team_id, is_user_id, is_identified, uuid) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *',
+            [createdAt.toISO(), JSON.stringify(properties), teamId, isUserId, isIdentified, uuid]
+        )
+        const personCreated = insertResult.rows[0] as Person
+        if (this.kafkaProducer) {
+            const data = {
+                created_at: castTimestampOrNow(createdAt),
+                properties: JSON.stringify(properties),
+                team_id: teamId,
+                is_identified: isIdentified,
+                id: uuid,
+            }
+            await this.kafkaProducer.send({
+                topic: KAFKA_PERSON,
+                messages: [{ value: Buffer.from(JSON.stringify(data)) }],
+            })
+        }
+        return personCreated
+    }
+
+    public async updatePerson(person: Person, update: Partial<Person>): Promise<Person> {
+        const updatedPerson: Person = { ...person, ...update }
+        await this.postgresQuery(
+            `UPDATE posthog_person SET ${Object.keys(update).map(
+                (field, index) => field + ' = $' + (index + 1)
```

Comment:
`field` here is passed to SQL without sanitisation. We have to be really really sure nobody starts passing arbitrary SQL in the keys of the `update` argument. 

There was/is already a case with passing bad data into `update` below in `processEvent`. Who knows who will be working on this codebase in the future and what this function will be called with. So we should escape or sanitize these fields somehow.

Reply:
Added `sanitizeSqlIdentifier` to delete quotes just in case